src/HeaderCell.ts

```diff
@@ -10,14 +10,16 @@ import * as cellCss from './styles/shared/cell.m.css';
 
 export const HeaderCellBase = ThemeableMixin(RegistryMixin(WidgetBase));
 
-export interface HeaderCellProperties extends ThemeableProperties, HasColumn, HasSortDetail, HasSortEvent, RegistryMixinProperties {}
+export interface HeaderCellProperties extends ThemeableProperties, HasColumn, HasSortDetail, HasSortEvent, RegistryMixinProperties {
+	key: string;
```

Comment:
All this is meant to do is to make key, which is optional in normal `WidgetProperties` a required field.